osu.Game.Rulesets.Taiko/Mods/TaikoModClassic.cs

```diff
@@ -3,21 +3,200 @@
+                Duration = original.Duration;
+                TickRate = original.TickRate;
+                RequiredGoodHits = original.RequiredGoodHits;
+                RequiredGreatHits = original.RequiredGreatHits;
+            }
+
+            public override Judgement CreateJudgement() => new TaikoClassicDrumRollJudgement();
+
+            protected override void CreateTicks(CancellationToken cancellationToken)
+            {
+                if (TickSpacing == 0)
+                    return;
+
+                bool first = true;
+
+                for (double t = StartTime; t < EndTime + TickSpacing / 2; t += TickSpacing)
+                {
+                    cancellationToken.ThrowIfCancellationRequested();
+
+                    AddNested(new ClassicDrumRollTick
+                    {
+                        FirstTick = first,
+                        TickSpacing = TickSpacing,
+                        StartTime = t,
+                        IsStrong = IsStrong
+                    });
+
+                    first = false;
+                }
+            }
+        }
+
+        private class ClassicDrumRollTick : DrumRollTick
+        {
+            public override Judgement CreateJudgement() => new TaikoClassicDrumRollTickJudgement();
+        }
+
+        private class ClassicSwell : Swell
+        {
+            public ClassicSwell(Swell original)
+            {
+                StartTime = original.StartTime;
+                Samples = original.Samples;
+                EndTime = original.EndTime;
+                Duration = original.Duration;
+                RequiredHits = original.RequiredHits;
+            }
+
+            public override Judgement CreateJudgement() => new TaikoClassicSwellJudgement();
+        }
+
+        private class TaikoClassicDrumRollJudgement : TaikoDrumRollJudgement
+        {
+            public override HitResult MaxResult => HitResult.LargeBonus;
+        }
+
+        private class TaikoClassicDrumRollTickJudgement : TaikoDrumRollTickJudgement
+        {
+            public override HitResult MaxResult => HitResult.LargeBonus;
+        }
+
+        private class TaikoClassicSwellJudgement : TaikoSwellJudgement
+        {
+            public override HitResult MaxResult => HitResult.LargeBonus;
+        }
+
+        private class ClassicDrawableDrumRoll : DrawableDrumRoll
+        {
+            public override bool DisplayResult => false;
+
+            protected override void CheckForResult(bool userTriggered, double timeOffset)
+            {
+                if (userTriggered)
+                    return;
+
+                if (timeOffset < 0)
+                    return;
+
+                ApplyResult(r => r.Type = HitResult.IgnoreMiss);
+            }
+        }
+
+        private class ClassicDrawableSwell : DrawableSwell
+        {
+            public override bool DisplayResult => false;
+
+            protected override void CheckForResult(bool userTriggered, double timeOffset)
```

Comment:
Not sure about overriding this one either. It's not strictly necessary. I would prefer keeping just the base implementation, and tweaking it a bit so that this code duplication can be removed.

- The following can be replaced with `r.Judgement.MaxResult` in the base:

   https://github.com/ppy/osu/blob/bc5e4fd6ca6e4b5e677013c6a36728bab717775f/osu.Game.Rulesets.Taiko/Objects/Drawables/DrawableSwell.cs#L204

- The worse one is this one:

  https://github.com/ppy/osu/blob/bc5e4fd6ca6e4b5e677013c6a36728bab717775f/osu.Game.Rulesets.Taiko/Objects/Drawables/DrawableSwell.cs#L225

  In this case I'd probably opt for exposing a `public virtual HitResult OkResult => HitResult.Ok` in the base class and overriding it here to `public override HitResult OkResult => HitResult.SmallBonus`.

As long as there are no logic changes in the actual judgement assignment process (and there don't seem to be, seems like a pure copy-paste), there should be no behavioural changes after applying the above.